diff --git a/src/multimedia/platform/qplatformaudiobufferinput_p.h b/src/multimedia/platform/qplatformaudiobufferinput_p.h
index 55636ce06..530266a4f 100644
--- a/src/multimedia/platform/qplatformaudiobufferinput_p.h
+++ b/src/multimedia/platform/qplatformaudiobufferinput_p.h
@@ -38,9 +38,9 @@ public:
     const QAudioFormat &audioFormat() const { return m_format; }
 
     QMediaInputEncoderInterface *encoderInterface() const { return m_encoderInterface; }
-    void setEncoderInterface(QMediaInputEncoderInterface *interface)
+    void setEncoderInterface(QMediaInputEncoderInterface *encoderInterface)
     {
-        m_encoderInterface = interface;
+        m_encoderInterface = encoderInterface;
     }
 
 Q_SIGNALS:
diff --git a/src/multimedia/platform/qplatformvideoframeinput_p.h b/src/multimedia/platform/qplatformvideoframeinput_p.h
index 45714492c..ccab7c4c5 100644
--- a/src/multimedia/platform/qplatformvideoframeinput_p.h
+++ b/src/multimedia/platform/qplatformvideoframeinput_p.h
@@ -37,9 +37,9 @@ public:
     QString errorString() const final { return {}; }
 
     QMediaInputEncoderInterface *encoderInterface() const { return m_encoderInterface; }
-    void setEncoderInterface(QMediaInputEncoderInterface *interface)
+    void setEncoderInterface(QMediaInputEncoderInterface *encoderInterface)
     {
-        m_encoderInterface = interface;
+        m_encoderInterface = encoderInterface;
     }
 
 Q_SIGNALS:
